ext/dom/html_document.c

```diff
@@ -511,7 +511,8 @@ static bool dom_decode_encode_fast_path(
 	const lxb_char_t *last_output = buf_ref;
 	while (buf_ref != buf_end) {
 		const lxb_char_t *buf_ref_backup = buf_ref;
-		lxb_codepoint_t codepoint = decoding_encoding_ctx->decode_data->decode_single(&decoding_encoding_ctx->decode, &buf_ref, buf_end);
+		/* Fast path converts non-validated UTF-8 -> validated UTF-8 */
+		lxb_codepoint_t codepoint = lxb_encoding_decode_utf_8_single(&decoding_encoding_ctx->decode, &buf_ref, buf_end);
```

Comment:
Complete aside, but is there anyway to use the information of the UTF-8 flags on zend_strings or not to by pass some of theses (although, this feels very unlikely most strings would be knowns to be valid UTF-8 in the first place)

Reply:
Interesting idea.
It's possible to do this for the `createFromString` method. How useful this can be useful in real-world cases, I don't know.